### ISSUE TITLE
After loading fine-tuned model from local and use it for prediction, it continue training from scratch again!

### ISSUE THREAD

**Issue by Fushier** [huggingface/transformers#12034]
transformers: 4.6.1
torch: 1.3
gpu: k40m * 2
datasets: msra_ner
model: hfl/chinese-bert-wwm

I'm fine-tuning a model for token classification, and after training, I save the model:
`trainer.save_model('./model')
trainer.save_metrics('./model')`

Now I load the saved model : 
`
tokenizer = AutoTokenizer.from_pretrained("./model")
config = transformers.AutoConfig.from_pretrained("./model")
model = AutoModelForTokenClassification.from_pretrained("./model", config=config)
args = TrainingArguments(
    output_dir='./results'
)

trainer = Trainer(
    model,
    args,
    data_collator=data_collator,
    tokenizer=tokenizer,
)
`

And then predict with test dataset:
`
predictions, labels, metrics = trainer.predict(tokenized_datasets)
`

But in terminal, I get:
Some weights of the model checkpoint at hfl/chinese-bert-wwm were not used when initializing BertForTokenClassification: ['cls.predictions.transform.dense.weight', 'cls.seq_relationship.weight', 'cls.seq_relationship.bias', 'cls.predictions.transform.dense.bias', 'cls.predictions.transform.LayerNorm.bias', 'cls.predictions.transform.LayerNorm.weight', 'cls.predictions.bias', 'cls.predictions.decoder.weight']
- This IS expected if you are initializing BertForTokenClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing BertForTokenClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
Some weights of BertForTokenClassification were not initialized from the model checkpoint at hfl/chinese-bert-wwm and are newly initialized: ['classifier.bias', 'classifier.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
  0%|                                                                                                                                 | 0/28130 [00:00<?, ?it/s]

It seems like that the model starts training again?

@sgugger

**Comment by sgugger** [comment]
No, this means you are loading a model that hasn't been fine-tuned. The model present in in your `./model` directory seems to have a pretraining head which is discarded, and a new token classification head is instantiated.
